### PR TITLE
pythonPackages.brotlicffi: fix for python < 3.6 the same way the upstream does

### PR DESCRIPTION
**Pull request by vojta001**
###### Motivation for this change

This library now works with both Python 2.7 and Python 3.9 (did simple tests with both versions)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/brotlicffi/default.nix [hunk 1]
```diff
@@ -24,7 +24,10 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [
     cffi
-  ];
+  ] ++ (if pythonOlder "3.4" then
+    [ enum34 ]
+  else
+    []);
```
Comment (jonringer) [inline pkgs/development/python-modules/brotlicffi/default.nix:30]: most common way of expressing this.
```suggestion
  ] ++ lib.optionals (pythonOlder "3.4") [
    enum34
  ];
```

Comment (vojta001) [reply pkgs/development/python-modules/brotlicffi/default.nix]: Thanks! Implemented

File: pkgs/development/python-modules/brotlicffi/default.nix [hunk 2]
```diff
@@ -4,12 +4,12 @@
 , pythonOlder
 , cffi
 , brotli
+, enum34
 }:
 
 buildPythonPackage rec {
   pname = "brotlicffi";
   version = "1.0.9.2";
-  disabled = pythonOlder "3.6";
```
Comment (fabaff) [inline pkgs/development/python-modules/brotlicffi/default.nix:12]: Why do we need Python 2 support? No package seems to need it. `requests` is the only consumer of `brotlicffi` and is using `brotli` for Python 2.

Python 2 is now EOL for 20 months.

Comment (r-burns) [reply pkgs/development/python-modules/brotlicffi/default.nix:12]: I'm not so sure that we care about python < 3.6 either. 3.5 is already EOL for over a year.

**Comment by vojta001** [comment]
There are still some applications that need the Python 2 machinery like
Natron (I know it's somewhat their problem).

And since brotlicffi explicitly has enum34 as a dependency for old Python
versions, it made sense to me to do it in nixpkgs as well. Otherwise we
could drop enum34 from nixpkgs completely.

**Comment by SuperSandro2000** [comment]
> Otherwise we could drop enum34 from nixpkgs completely.

I personally would prefer this so that we get rid more and more of python2 and older python3s.

**Comment by vojta001** [comment]
I am afraid enum34 is still used by too many packages.

I'll close this PR however, as Natron, my primary motivation, expects to release a Python3 supporting version soon.

**Pull request closed without merging**